### PR TITLE
Forgot password functionality

### PR DESCRIPTION
**Pull request by elsa17nam**
Added functionality for forgot password. Previously, nothing happens when you click Reset Password. In this new changes, it checks whether the email exists or not. If existing, it will be redirected to resetPasswordPage where user may enter new password and confirm password. Then when you click Reset Password in resetPasswordPage, you will be redirected to login page.

### CODE REVIEW

**Comment by fencim** [comment]
allow me to fix the build error

File: src/stores/profile.store.ts [hunk 1]
```diff
@@ -18,15 +18,12 @@ export const useProfileStore = defineStore('profile', {
...
     },
-    async findProfiles(keyword: string) {
-      return ((await firebaseService.findAll('profiles')) as IProfile[]).filter(
-        (profile) => {
-          return (
-            profile.type == 'adviser' &&
-            profile.fullName.toLowerCase().match(keyword.toLowerCase())
-          );
-        }
-      );
```
Comment (fencim) [inline src/stores/profile.store.ts]: avoid removing others code of ther developers

File: src/pages/auth/resetPasswordPage.vue
Comment (fencim) [inline src/pages/auth/resetPasswordPage.vue:1]: Follow PascalCasing on file name `ResetPasswordPage.vue`

File: src/pages/auth/ForgotPasswordPage.vue [hunk 2]
```diff
@@ -41,45 +41,60 @@
...
 };
 
-// Navigate to login page
-const navigateToLogin = () => {
-  router.push('/auth/login'); // Adjust the route if necessary
-};
+// Example email validation function
+const validateEmail = (email) => {
+  const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return emailPattern.test(email);
```
Comment (fencim) [inline src/pages/auth/ForgotPasswordPage.vue]: this causing build failure, you may want use automatic email validation of q-form via q-input rules

File: src/pages/auth/ForgotPasswordPage.vue [hunk 3]
```diff
@@ -41,45 +41,60 @@
...
   Notify.create({
     type: 'positive',
-    message: 'This is a placeholder for the password reset functionality.',
+    message: 'Password reset link has been sent. Redirecting...',
   });
 
-  console.log(`Attempted password reset for email: ${emailTrimmed}`);
+  // Redirect to ResetPasswordPage.vue
+  setTimeout(() => {
+    router.push('/auth/reset-password');
```
Comment (fencim) [inline src/pages/auth/ForgotPasswordPage.vue]: when routing better use name of routes

**Review by fencim — CHANGES_REQUESTED** [review]

File: src/stores/profile.store.ts (hunk 1 above)
Comment (eyesoreee) [reply src/stores/profile.store.ts]: I apologize, sir. That was my mistake. I was the one who advised Ate Intishar (@elsa17nam) to modify, or in this case, remove it. It won't happen again. I simply overlooked it.

**Review by fencim — APPROVED** [review]

File: src/stores/profile.store.ts (hunk 1 above)
Comment (fencim) [reply src/stores/profile.store.ts]: I reverted it as it is being used in SearchBarComponent

**Comment by fencim** [comment]
merging to Let QA, to test on production site

**Pull request merged**